From 1e070eaef6c2ed4a4f589e4722ee99442e083ab1 Mon Sep 17 00:00:00 2001
From: Matt Selsky <matthew.selsky@twosigma.com>
Date: Fri, 2 Dec 2016 00:21:27 -0500
Subject: [PATCH 115/268] Compiler warning cleanup

warning: initializing 'char *' with an expression of type 'const char [10]' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
---
 ntpd/keyword-gen.c  |  6 +++---
 ntpd/refclock_jjy.c | 12 ++++++------
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/ntpd/keyword-gen.c b/ntpd/keyword-gen.c
index 6be4b50..d4533a9 100644
--- a/ntpd/keyword-gen.c
+++ b/ntpd/keyword-gen.c
@@ -15,7 +15,7 @@
 
 /* Define a structure to hold a (keyword, token) pair */
 struct key_tok {
-	char *	key;		/* Keyword */
+	const char * key;	/* Keyword */
 	u_short	token;		/* Associated Token */
 	follby	followedby;	/* nonzero indicates the next token(s)
 				   forced to be string(s) */
@@ -245,7 +245,7 @@ static void	generate_preamble	(void);
 static void	generate_fsm		(void);
 static void	generate_token_text	(void);
 static u_short	create_keyword_scanner	(void);
-static u_short	create_scan_states	(char *, u_short, follby, u_short);
+static u_short	create_scan_states	(const char *, u_short, follby, u_short);
 int		compare_key_tok_id	(const void *, const void *);
 int		compare_key_tok_text	(const void *, const void *);
 void		populate_symb		(char *);
@@ -468,7 +468,7 @@ generate_fsm(void)
  */
 static u_short
 create_scan_states(
-	char *	text, 
+	const char * text,
 	u_short	token, 
 	follby	followedby,
 	u_short	prev_state
diff --git a/ntpd/refclock_jjy.c b/ntpd/refclock_jjy.c
index 45b8641..a67e1d1 100644
--- a/ntpd/refclock_jjy.c
+++ b/ntpd/refclock_jjy.c
@@ -126,7 +126,7 @@
  */
 
 struct jjyRawDataBreak {
-	char	*pString ;
+	const char	*pString ;
 	int 	iLength ;
 } ;
 
@@ -3223,7 +3223,7 @@ static int
 teljjy_login_login ( struct peer *peer, struct refclockproc *pp, struct jjyunit *up )
 {
 
-	char	*pCmd ;
+	const char	*pCmd ;
 	int	iCmdLen ;
 
 	UNUSED_ARG(peer);
@@ -4046,7 +4046,7 @@ static int
 modem_init_resp00 ( struct peer *peer, struct refclockproc *pp, struct jjyunit *up )
 {
 
-	char	*pCmd, cBuf [ 46 ] ;
+	const char	*pCmd, cBuf [ 46 ] ;
 	int	iCmdLen ;
 	int	iErrorCorrection, iSpeakerSwitch, iSpeakerVolume ;
 	int	iNextModemState = STAY_MODEM_STATE ;
@@ -4324,7 +4324,7 @@ modem_esc_escape ( struct peer *peer, struct refclockproc *pp, struct jjyunit *u
 {
 	UNUSED_ARG(up);
 
-	char	*pCmd ;
+	const char	*pCmd ;
 	int	iCmdLen ;
 
 	DEBUG_MODEM_PRINTF( "modem_esc_escape" ) ;
@@ -4391,7 +4391,7 @@ static int
 modem_esc_disc ( struct peer *peer, struct refclockproc *pp, struct jjyunit *up )
 {
 
-	char	*pCmd ;
+	const char	*pCmd ;
 	int	iCmdLen ;
 
 	UNUSED_ARG(up);
@@ -4426,7 +4426,7 @@ jjy_write_clockstats ( struct peer *peer, int iMark, const char *pData )
 {
 
 	char	sLog [ 100 ] ;
-	char	*pMark ;
+	const char *pMark ;
 	int 	iMarkLen, iDataLen ;
 
 	switch ( iMark ) {
-- 
2.7.4

